shared/js/content-scripts/block-cookie.js

```diff
@@ -0,0 +1,51 @@
+// Set up 3rd party cookie blocker
+(function cookieBlocker () {
+    // don't inject into non-HTML documents (such as XML documents)
+    // but do inject into XHTML documents
+    if (document instanceof HTMLDocument === false && (
+        document instanceof XMLDocument === false ||
+        document.createElement('div') instanceof HTMLDivElement === false
+    )) {
+        return
+    }
+
+    function clearInjectedCookiesAndBlock () {
+        // Clear previously set cookies
+        var cookies = document.cookie.split('; ')
+        for (var c = 0; c < cookies.length; c++) {
+            var d = window.location.hostname
+            var cookieBase = encodeURIComponent(cookies[c].split(';')[0].split('=')[0]) + '=; expires=Thu, 01-Jan-1970 00:00:01 GMT; domain=' + d + ' ;path='
+            var p = location.pathname.split('/')
+            document.cookie = cookieBase + '/'
+            while (p.length > 0) {
+                document.cookie = cookieBase + p.join('/')
+                p.pop()
+            };
+        }
```

Comment:
It occurs to me that this may cause issues for sites that the user also visits in first party context. Consider a site that embeds a 3rd party iframe from YouTube. This code will wipe out all cookies except those that are http-only. On a subsequent visit to Youtube, there may be some breakage caused by the absense of these cookies.

Perhaps it is sufficient just to block the cookie setter and getter in this case?

Reply:
Hmm this is a good point. What are your thoughts @kdzwinel?

Reply:
Hm, good point. Those cookies will be still saved in a browser but should not be shared in the third party context. My only worry is that, as far as I remember, the removal of all 3p cookies was helping us to avoid the timing issue (i.e. if we injected our script after some cookies were already set it's OK because we would remove them). Am I recalling this correctly? Is this still the case?

Reply:
Yes this is still here to take care of any race conditions

Reply:
I still think this will cause breakage. For example, on YouTube there is a consent cookie set as not http-only, which would therefore be wiped by this. This means that everytime you visit a page with embedded youtube iframe, your consent settings would get wiped.

I think fixes for race conditions are planned, so we can wait for a proper solution there before addressing this.

Reply:
Hm, OK lets drop removing cookies for now. We have to note this gap in the relevant task and update tech design.